File: src/ServerPulse.Frontend/src/app/modules/authentication/domain/dtos/loginAccessTokenData.ts
import { AccessTokenData } from "../..";

export interface LoginAccessTokenData {
    accessToken: string;
    refreshToken: string;
    refreshTokenExpiryDate: Date;
}

export function mapLoginAccessTokenDataToAuthToken(response: LoginAccessTokenData): AccessTokenData {
    return {
        accessToken: response?.accessToken,
        refreshToken: response?.refreshToken,
        refreshTokenExpiryDate: new Date(response?.refreshTokenExpiryDate)
    }
}

File: src/ServerPulse.Backend/AuthenticationApi.Core/Dtos/Endpoints/Auth/Login/LoginAccessTokenData.cs
namespace AuthenticationApi.Core.Dtos.Endpoints.Auth.Login
{
    public class LoginAccessTokenData
    {
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime? RefreshTokenExpiryDate { get; set; }
    }
}
